jenkins-integration/beaker/install/shared/collect_facter_and_scenario_data.rb

```diff
@@ -0,0 +1,59 @@
+# Collect artifacts from machines
+# Brian Cain {brian.cain@puppetlabs.com}
+
+require 'puppet/gatling/config'
+require 'csv'
+require 'json'
+
+def get_facter_data
+  result = on master, "facter -p -j"
+  result.stdout
+end
+
+def save_data(facter_data, data_hash, simulation_id, config)
+  simulation_dir = File.join("..", "puppet-gatling", simulation_id)
+  unless File.exist?(simulation_dir)
+    FileUtils.mkdir_p simulation_dir
+  end
+
+  File.open(File.join(simulation_dir, 'facter-data.json'), 'w') { |file| file.write(facter_data) }
+  CSV.open(File.join(simulation_dir, 'important_data.csv'), 'w') { |csv| data_hash.to_a.each { |elem| csv << elem} }
+
+  File.open(File.join(simulation_dir, 'gatling_sim_data.csv'), 'w') do |file|
+    config["nodes"].each do |node|
+      file.write("#{File.basename(node["node_config"], ".json")},#{node["num_instances"]},#{node["num_repetitions"]}\n")
+    end
+  end
+  puts "Files now saved within #{simulation_dir}"
+end
+
+# disk, num cpus, speed of cpus, ram
+def get_data_hash(data)
+  facts_data = JSON.parse(data)
+  {
+      'processor0' => facts_data['processors']['models'][0],
+      'processorcount' => facts_data['processors']['count'],
+      'puppetversion' => facts_data['puppetversion'],
+      'blockdevice_sda_size' => facts_data['disks']['sda']['size'],
+      'memorysize' => facts_data['memory']['system']['total']
+  }
+end
+
+# Begin work
+
+puts "Gathering facter data and processing data..."
```

Comment:
This is a good point, we should probably try to stick to the Beaker::Log one.  Will try to clean that up later.